ark-core/src/vtxo.rs

```diff
@@ -320,3 +320,294 @@ where
 
     Ok(VirtualTxOutpoints { spendable, expired })
 }
+
+/// A virtual UTXO script that represents a collection of taproot scripts
+/// forming a taproot tree. This is equivalent to the TypeScript VtxoScript class.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct VirtualUtxoScript {
```

Comment:
I think we should move all this new code out of `vtxo.rs` and maybe into `arknote.rs`?